static/app/components/charts/worldMapChart.tsx

```diff
@@ -46,136 +37,118 @@ const DEFAULT_CENTER_X = 10.97;
 const DISCOVER_QUERY_LIST_CENTER_Y = -12;
 const DEFAULT_CENTER_Y = 9.71;
 
-class WorldMapChart extends React.Component<Props, State> {
-  state: State = {
-    countryToCodeMap: null,
-    map: null,
-    codeToCountryMap: null,
-  };
+export function WorldMapChart({
+  series,
+  seriesOptions,
+  fromDiscover,
+  fromDiscoverQueryList,
+  ...props
+}: WorldMapChartProps) {
+  const map = worldMap;
+  const codeToCountryMap = Object.fromEntries(
+    Object.entries(countryToCodeMap).map(([country, code]) => [code, country])
+  );
 
-  async componentDidMount() {
-    const [countryToCodeMap, worldMap] = await Promise.all([
```

Comment:
the way we are currently rendering this is causing some side effects, such as "Can't perform a react state update..."